### PR TITLE
Add search radius functionality

### LINKED ISSUE #33: Add a search radius to the geo and Postcode search
The search feature can get relevant results near to your location (either through Postcode entry or using the Geolocation API). But you can't change the search radius, which is currently hardcoded in the Lambda function. Need to add an input below the Postcode search for radius entry with a sensible default.

Postcode search is coming soon to the list view, in #29.

### PR DESCRIPTION
**Pull request by Sacharified**
Fixes #33 

This adds a search radius input, which accepts a number in Km and triggers a search with a new `radius` query parameter.

New input

The lambda now accepts a `radius` query parameter.

The default radius is infinity because that's the default in the `sphere-knn` package and results in the same behaviour as what's running in prod right now.

### CODE REVIEW

File: src/components/PostcodeSearch/index.js [hunk 1]
```diff
@@ -23,6 +24,14 @@ function PostcodeSearch() {
     }
   };
 
+  const handleRadiusChange = (e) => {
+    const {value} = e.currentTarget
+    
+    setRadius(value);
+    const valueKm = parseInt(value) * 1000;
+    setRadiusOnContext(valueKm);
```
Comment (Sacharified) [inline src/components/PostcodeSearch/index.js:37]: Happy to receive any feedback on how to improve the interaction between this form and the GeoContext

**Comment by tomoakley** [comment]
thanks @Sacharified!

Couple of things:
- if it's showing all results, and I type in the search radius could it switch to geo mode? and maybe show an error if it's not able to
- I think a default would be useful - e.g 15km. The label at the top of the list ("showing x results closest to you") could then be updated to "showing x results within 15km of you". In fact, moving the search radius input to be inline with that label might work quite well - but only showing it on geo/postcode mode; the label could then be "showing x results within [input box value] km of you". This negates the point I made above (because it wouldn't be visible in non-geo/postcode mode).

Let me know what you think of those suggestions.

**Comment by Sacharified** [comment]
> moving the search radius input to be inline with that label

@tomoakley Do you mean moving it out of the postcode search and in to the list component?

Agreed about the default, I'll update that.

**Comment by tomoakley** [comment]
> > moving the search radius input to be inline with that label
> 
> @tomoakley Do you mean moving it out of the postcode search and in to the list component?
> 
> Agreed about the default, I'll update that.

I was thinking as I typed there.  Not sure it's a good idea, but I think it'd be useful to not be able to input a search radius when seeing all the results (i.e no postcode,  no geo)

**Comment by Sacharified** [comment]
> > > moving the search radius input to be inline with that label
> > 
> > 
> > @tomoakley Do you mean moving it out of the postcode search and in to the list component?
> > Agreed about the default, I'll update that.
> 
> I was thinking as I typed there. Not sure it's a good idea, but I think it'd be useful to not be able to input a search radius when seeing all the results (i.e no postcode, no geo)

Yeah that makes sense.

So I could either hide it, disable it or automatically switch to Geo mode when it's typed into. Do you have any preference?

**Pull request closed without merging**